Content.Server/Paper/PaperSystem.cs

```diff
@@ -100,6 +111,44 @@ private void OnExamined(EntityUid uid, PaperComponent paperComp, ExaminedEvent a
             }
         }
 
+        private void UpdateTagsState(EntityUid userUid, PaperComponent paperComp)
+        {
+            if (paperComp.TagsState != null)
```

Comment:
```cs
if (paperComp.TagsState is not {} state)
    return;
```

then assign TagsState to it after everything is set since its a struct

Reply:
What kind of construction is this?